src/actions/recomputeReduxState.js

```diff
@@ -917,10 +917,37 @@ export const createStateFromQueryOrJSONs = ({
 
   /* Special handling of measurements controls and query params */
   if (measurements.loaded) {
-    const { collectionToDisplay, collectionControls, updatedQuery} = combineMeasurementsControlsAndQuery(measurements, query);
+    const {
+      collectionToDisplay,
+      collectionControls,
+      updatedQuery,
+      newColoringData,
+    } = combineMeasurementsControlsAndQuery(measurements, query);
     measurements.collectionToDisplay = collectionToDisplay;
     controls = {...controls, ...collectionControls};
     query = updatedQuery;
+
+    // Similar to the state changes applied for `ADD_EXTRA_METADATA`
+    if (newColoringData !== undefined) {
+      // Update controls
+      newColoringData.coloringsPresentOnTree.forEach((coloring) => controls.coloringsPresentOnTree.add(coloring));
+      // Update metadata
+      metadata.colorings = {...metadata.colorings, ...newColoringData.colorings};
+      // Update tree
+      addNodeAttrs(tree.nodes, newColoringData.nodeAttrs);
+      Object.keys(newColoringData.colorings).forEach((attr) => tree.nodeAttrKeys.add(attr));
+      const nonContinuousColorings = Object.keys(newColoringData.colorings).filter((coloring) => {
+        return newColoringData.colorings[coloring].type !== "continuous"
+      });
+      tree.totalStateCounts = {
+        ...tree.totalStateCounts,
+        ...countTraitsAcrossTree(tree.nodes, nonContinuousColorings, false, true)
+      };
+      // Update treeToo if exists
+      if (treeToo && treeToo.loaded) {
+        addNodeAttrs(treeToo.nodes, newColoringData.nodeAttrs);
+      }
```

Comment:
Can we have two trees + measurements? I've come to the opinion that we should drop a lot of functionality when viewing two trees, so that if the second tree is displayed we don't allow measurements, frequencies, map to be displayed.

Reply:
Two trees + measurements is definitely possible right now, where the measurements panel always uses the measurements JSON from the main tree. 

I'm not entirely sure how useful it is though...could it be useful for HA:NA + measurements?

Reply:
> I'm not entirely sure how useful it is though...could it be useful for HA:NA + measurements?

I can't answer that, but we should find the answer. Two trees complicates Auspice quite a lot, and I think it would be a worthwhile push to limit the cases in order to simplify the code.

Reply:
We can revisit simplifying code in the future, starting conversation in [Slack](https://bedfordlab.slack.com/archives/C03KWDET9/p1737491902430339).